source/MRViewer/MRRibbonNotification.cpp

```diff
@@ -32,18 +32,127 @@ namespace MR
 
 void RibbonNotifier::pushNotification( const RibbonNotification& notification )
 {
-    if ( !notifications_.empty() && notifications_.front().notification == notification )
+    addNotification_( notifications_, notification );
+    addNotification_( notificationsHistory_, notification );
+
+    highestNotification_ = NotificationType::Count;
+    for ( const auto& nwt : notificationsHistory_ )
+    {
+        if ( int( nwt.notification.type ) < int( highestNotification_ ) )
+            highestNotification_ = nwt.notification.type;
+    }
+    
+    requestClosestRedraw_();
+}
+
+void RibbonNotifier::drawNotificationHistoryButton( float scaling )
+{
+    if ( !notifications_.empty() || notificationsHistory_.empty() )
+        return;
+
+    float notificationsPosX = 0.f;
+    if ( auto menu = getViewerInstance().getMenuPluginAs<RibbonMenu>() )
+        notificationsPosX = float( menu->getSceneSize().x );
+    Vector2f currentPos = Vector2f( notificationsPosX, float( getViewerInstance().framebufferSize.y ) - 25.f * scaling );
+    const Vector2f padding = Vector2f( 0.0f, 20.0f * scaling );
+
+    
+    auto iconsFont = RibbonFontManager::getFontByTypeStatic( RibbonFontManager::FontType::Icons );
+    if ( iconsFont )
+    {
+        iconsFont->Scale = 0.7f;
+        ImGui::PushFont( iconsFont );
+    }
+    const float size = ImGui::GetTextLineHeight() + 10.f * scaling * 2.f;
+    if ( iconsFont )
+    {
+        iconsFont->Scale = 1.0f;
+        ImGui::PopFont();
+    }
+    
+    currentPos.x += padding.y;
+
+    currentPos -= padding;
+
+    ImGui::SetNextWindowPos( currentPos, ImGuiCond_Always, ImVec2( 0.f, 1.0f ) );
+    ImGui::SetNextWindowSize( ImVec2( size, size ), ImGuiCond_Always );
+    ImGuiWindowFlags flags =
+        ImGuiWindowFlags_AlwaysAutoResize |
+        ImGuiWindowFlags_NoResize |
+        ImGuiWindowFlags_NoTitleBar |
+        ImGuiWindowFlags_NoMove;
+    std::string name = "##NotificationButton";
+    ImGui::PushStyleVar( ImGuiStyleVar_WindowBorderSize, 0.0f );
+    ImGui::PushStyleVar( ImGuiStyleVar_WindowRounding, 4.0f * scaling );
+    ImGui::PushStyleVar( ImGuiStyleVar_WindowPadding, ImVec2( 10.0f * scaling, 10.0f * scaling ) );
+    ImGui::PushStyleVar( ImGuiStyleVar_CellPadding, { 0, 0 } );
+    ImGui::PushStyleColor( ImGuiCol_WindowBg, MR::ColorTheme::getRibbonColor( MR::ColorTheme::RibbonColorsType::FrameBackground ).getUInt32() );
+
+    auto activeModal = ImGui::GetTopMostPopupModal();
+
+    ImGui::Begin( name.c_str(), nullptr, flags );
+
+    if ( ImGui::IsWindowAppearing() )
+    {
+        if ( !activeModal )
+            ImGui::BringWindowToDisplayFront( ImGui::GetCurrentWindow() ); // bring to front to be over modal background
+        if ( !ProgressBar::isOrdered() && !activeModal ) // do not focus window, not to close modal on appearing
+            ImGui::SetWindowFocus();
+    }
```

Comment:
it is not needed for "Notification History" it is simple button that should not appear over other windows

Reply:
fixed